### ISSUE TITLE
llama.h2o.ai and docker not 'NoneType' when using "RelSources" or "TopKSources"

### ISSUE THREAD

**Issue by dingzhongshan** [h2oai/h2ogpt#996]
Good afternoon,
First, thank @pseudotensor and team for all your hard work on bringing this disparate set of libraries together for the community. 

I recently upgrade to the docker gcr.io/vorvan/h2oai/h2ogpt-runtime (build_date="20231024")  to test and then confirmed the same errors on llama.h2o and gpt.h2o.ai.

I believe I have the defaults setup (see screencap) per the FAQ and DOC, but please correct me if I missed a recent change.
Problem:  Using "RelSources" and "TopKSources" with Query, Summary or Extract yields the error:

`int() argument must be a string, a bytes-like object or real number, not 'None Type'`

Setup (screens attached)
1. Select "MyData" 
2. Upload 2 paul graham essays; "1paul_graham.txt" and "2paul_graham.txt" (16K & 4K tokens)
3. Confirm upload in "Document Viewer"
4. Selected "document" under "Select Subset of Documents" drop down. 
5. Select "Relevant" + [Query, Summarize, Extract] to test (no problem)
6. Select "RelSources" + [Query, Summarize, Extract] to test (error above)
7. Select "TopKSources" + [Query, Summarize, Extract] (error above)

Trace from the docker container:
![stack](https://github.com/h2oai/h2ogpt/assets/217438/17608eb2-498a-4d7e-b22b-2899e199a9c3)
[build_version.txt](https://github.com/h2oai/h2ogpt/files/13127674/build_version.txt)

Screenshots from llama.h2o.ai:
![0](https://github.com/h2oai/h2ogpt/assets/217438/3ac79103-a476-49a1-8292-29866992d169)
![1](https://github.com/h2oai/h2ogpt/assets/217438/71abc519-2c56-48cc-ab16-7d3336414f5f)
![2](https://github.com/h2oai/h2ogpt/assets/217438/1ca70f96-f9ef-4465-b443-5a5474dce363)
![3](https://github.com/h2oai/h2ogpt/assets/217438/e6ea81e6-4668-45f9-90c3-03c4320bfa5b)

Additionally, when using llama.h2o.ai I ran into this error as well.  I didn't find the exact code block that prevents the token overrun, but I thought I saw it or maybe this is designed.
![4](https://github.com/h2oai/h2ogpt/assets/217438/bb9f7772-e8bd-4677-a85f-d791a17e1099)
![5](https://github.com/h2oai/h2ogpt/assets/217438/5a17c723-ac0b-47e5-890e-c5b68c684af1)

Thanks (ding).

**Comment by pseudotensor** [comment]
For the token error, are you able to share a public prompt/document that does that?  I'm trying to make it hit that, but if you could share details that would be easiest.

**Comment by dingzhongshan** [comment]
Absolutely.  For the token error: 
* after the document (attached) was upload and selected, I also selected mydata, select relevant/rel or topk, and selected summarize or extract

I tried three different prompts (no changes to prompts or settings under expert tab):
1.   blank/no prompt (just submit per the text box "For sum
[1paul_graham.txt](https://github.com/h2oai/h2ogpt/files/13134343/1paul_graham.txt)
marization, no query required, just click submit" 
2. `What did Paul Graham do before college?`
3. `According to the selected document, what did Paul Graham do before college?`
